packages/accordion/src/Accordion.ts

```diff
@@ -91,23 +87,22 @@ export class Accordion extends SizedMixin(SpectrumElement, {
         });
     }
 
-    private handleSlotchange(): void {
+    private handleSlotchange({
+        target: slot,
+    }: Event & { target: HTMLSlotElement }): void {
         this.focusGroupController.clearElementCache();
-    }
 
-    private setAccordionProperties(): void {
-        this.items.forEach((item) => {
+        const assignedElements = slot.assignedElements() as AccordionItem[];
+        assignedElements.forEach((item) => {
             item.size = this.size;
         });
     }
 
     protected override render(): TemplateResult {
-        this.setAccordionProperties();
```

Comment:
@vjosyula Setting property values during render can cause cycles so it is typically avoided or not allowed.  In this case, the setAccordionProperties would be setting properties on the slotted items at render time which is not ideal. I moved this work up into the slot change handler so that it can process at the time the slotted item is registered. Let me know if this works for you @najikahalsema!